### PR TITLE
Mindy/discover episodes bug

### LINKED ISSUE #398: Discover showing episodes twice
These episodes reappear when scrolling in discover 

img_fa4e91a6b8f0-2
img_fa4e91a6b8f0-1

### PR DESCRIPTION
**Pull request by mindylou**
- fixes #398 by only reloading data on view will appear (instead of previously re-fetching everything)
- adds pull to refresh in discover

### CODE REVIEW

File: Podcast/DiscoverComponentViewController.swift [hunk 1]
```diff
@@ -62,4 +67,13 @@ class DiscoverComponentViewController: ViewController, NVActivityIndicatorViewab
         return header
     }
 
+    @objc func pullToRefresh() {
```
Comment (doctor3w) [inline Podcast/DiscoverComponentViewController.swift]: Why can't we just override this method?

File: Podcast/DiscoverTopicViewController.swift [hunk 2]
```diff
@@ -175,16 +177,21 @@ class DiscoverTopicViewController: DiscoverComponentViewController {
         navigationController?.navigationBar.setBackgroundImage(nil, for: .default)
     }
 
-    func configureTopic() {
+    func configureTopic(isPullToRefresh: Bool = false) {
```
Comment (doctor3w) [inline Podcast/DiscoverTopicViewController.swift]: Better name: canPullToRefresh. Also any reason we need the word topic in this function name? What about just "setup" or "configure".

File: Podcast/DiscoverViewController.swift [hunk 3]
```diff
@@ -121,11 +121,15 @@ class DiscoverViewController: DiscoverComponentViewController {
...
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        fetchDiscoverElements()
+        topSeriesCollectionView.reloadData()
+        topEpisodesTableView.reloadData()
     }
 
-    func fetchDiscoverElements() {
-        topSeriesCollectionView.reloadData()
+    func fetchDiscoverElements(isPullToRefresh: Bool = false) {
```
Comment (doctor3w) [inline Podcast/DiscoverViewController.swift]: Same here.

**Review by doctor3w — CHANGES_REQUESTED** [review]

File: Podcast/DiscoverComponentViewController.swift (hunk 1 above)
Comment (mindylou) [reply Podcast/DiscoverComponentViewController.swift]: i had that before and it was throwing an error before but now it's not... weird

File: Podcast/DiscoverTopicViewController.swift [hunk 4]
```diff
@@ -220,18 +226,20 @@ class DiscoverTopicViewController: DiscoverComponentViewController {
         relatedTopicsView.collectionView.reloadData()
     }
 
-    func fetchEpisodes(id: Int) {
+    func fetchEpisodes(id: Int, canPullToRefresh: Bool = false) {
 
         let topEpisodesForTopicEndpointRequest = DiscoverTopicEndpointRequest(requestType: .episodes, topicID: id, offset: offset, max: pageSize)
```
Comment (natashaarmbrust) [inline Podcast/DiscoverTopicViewController.swift]: let's reset the offset here instead of setup because this endpoint is what uses the offset

**Review by natashaarmbrust — COMMENTED** [review]
LGTM

**Pull request merged**